app/src/main/java/com/hedvig/app/feature/chat/data/ChatEventStore.kt

```diff
@@ -0,0 +1,34 @@
+package com.hedvig.app.feature.chat.data
+
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.intPreferencesKey
+import e
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.map
+
+interface ChatEventStore {
```

Comment:
Are we getting any mileage out of segregating the `interface` here? 🤔 Usually, I find that `mockk` can replicate the interface of any given `class` without any real hassle, and as such I've mostly skipped extracting an interface, unless I have a few underlying implementations that I want to support 🙂

Reply:
I suppose this is more future proofing, if we want to change the implementation later its nice to have an interface in place. Also nice if we want to inject debug implementations.